### PR TITLE
REF: avoid using internals methods for to_timestamp, to_period

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Review by simonjayhawkins — COMMENTED** [review]
@jbrockmendel while refactoring, may be worth adding annotations to to_timestamp and to_period

File: pandas/core/frame.py [hunk 1]
```diff
@@ -8299,23 +8295,21 @@ def to_period(self, freq=None, axis=0, copy=True) -> "DataFrame":
...
         -------
-        TimeSeries with PeriodIndex
+        DataFrame with PeriodIndex
         """
-        new_data = self._data
-        if copy:
-            new_data = new_data.copy()
+        new_obj = self.copy(deep=copy)
 
         axis = self._get_axis_number(axis)
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/frame.py]: i like it, will update

File: pandas/core/frame.py [hunk 2]
```diff
@@ -8265,26 +8265,22 @@ def to_timestamp(self, freq=None, how="start", axis=0, copy=True) -> "DataFrame"
...
+            new_obj.columns = self.columns.to_timestamp(freq=freq, how=how)
         else:  # pragma: no cover
             raise AssertionError(f"Axis must be 0 or 1. Got {axis}")
 
-        return self._constructor(new_data)
+        return new_obj
 
     def to_period(self, freq=None, axis=0, copy=True) -> "DataFrame":
         """
-        Convert DataFrame from DatetimeIndex to PeriodIndex.
```
Comment (simonjayhawkins) [inline pandas/core/frame.py]: I think you should undo this removal. (PEP257).

Comment (jbrockmendel) [reply pandas/core/frame.py]: ok, though im on record as disliking this sentence fragment

File: pandas/core/frame.py [hunk 3]
```diff
@@ -8265,23 +8265,16 @@ def to_timestamp(self, freq=None, how="start", axis=0, copy=True) -> "DataFrame"
...
-            assert isinstance(self.columns, (ABCDatetimeIndex, ABCPeriodIndex))
-            new_data.set_axis(0, self.columns.to_timestamp(freq=freq, how=how))
-        else:  # pragma: no cover
-            raise AssertionError(f"Axis must be 0 or 1. Got {axis}")
+        axis_name = self._get_axis_name(axis)
+        old_ax = getattr(self, axis_name)
+        new_ax = old_ax.to_timestamp(freq=freq, how=how)
 
-        return self._constructor(new_data)
+        setattr(new_obj, axis_name, new_ax)
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]: could use `set_axis(..., axis=)` instead?

Comment (jbrockmendel) [reply pandas/core/frame.py]: either way works

**Review by jreback — APPROVED** [review]
lgtm. @jorisvandenbossche comment, be either way is ok. merge away.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**